test/utils/update/beneficial_owners_managing_officers_data_fetch.spec.ts

```diff
@@ -43,5 +55,69 @@ describe("util beneficial owners managing officers data fetch", () => {
     retrieveBoAndMoData(req, appData);
     expect(mockGetCompanyPscService).not.toHaveBeenCalled();
     expect(mockGetCompanyOfficers).not.toHaveBeenCalled();
+    expect(mockGetManagingOfficerPrivateData).not.toHaveBeenCalled();
+  });
+
+  test("retrieveBoAndMoData sets MO private data in appData", async () => {
+    appData = { "transaction_id": "123", "overseas_entity_id": "456" };
+    mockGetCompanyPscService.mockReturnValue(MOCK_GET_COMPANY_PSC_ALL_BO_TYPES);
+    mockGetCompanyOfficers.mockReturnValue(MOCK_GET_COMPANY_OFFICERS);
+    mockGetManagingOfficerPrivateData.mockReturnValue(MOCK_GET_MO_PRIVATE_DATA);
+    await retrieveBoAndMoData(req, appData);
+    expect(appData.update?.review_managing_officers_individual?.length).toEqual(1);
+    const usual_residential_address = appData.update?.review_managing_officers_individual?.[0].usual_residential_address;
+    expect(usual_residential_address?.property_name_number).toEqual("URA 1");
+    expect(usual_residential_address?.line_1).toEqual("URA Address line 1");
+    expect(usual_residential_address?.line_2).toEqual("URA Address line 2");
+    expect(usual_residential_address?.town).toEqual("URA Town");
+    expect(usual_residential_address?.country).toEqual("URA Country");
+    expect(usual_residential_address?.county).toEqual("URA County");
+    expect(usual_residential_address?.postcode).toEqual("URA Postcode");
+    expect(appData.update?.bo_mo_data_fetched).toBe(true);
+  });
+
+  test("Should not call getManagingOfficerPrivateData when transactionId and overseasEntityId are undefined", async () => {
+    appData = { "transaction_id": undefined, "overseas_entity_id": undefined, "entity_number": "1234" };
+
+    mockGetCompanyOfficers.mockReturnValue(Promise.resolve({ items: [] }));
```

Comment:
This should return something shouldn't it to satisfy the test? Otherwise we don't get past line 69 in the `beneficial_owners_managing_officers_data_fetch.ts` class

Reply:
Misnamed. Added another test also.